### PR TITLE
feat: My Collection Edit artwork button

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [CX-2710]

## Description

This PR adds the My Collection Edit artwork button, adds `category` to the artwork attributes to be saved (Ooooops!) and adjusts some links and redirects depending on whether the user edits or creates an artwork.

[CX-2710]: https://artsyproduct.atlassian.net/browse/CX-2710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 1]
```diff
@@ -50,10 +50,12 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
   const { createOrUpdateArtwork } = useCreateOrUpdateArtwork()
 
   const handleSubmit = async (values: ArtworkModel) => {
+    console.log({ values })
     try {
       const artworkId = await createOrUpdateArtwork({
         artworkId: artwork?.internalID,
         artistIds: [values.artistId],
+        category: values.category,
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:58]: Very important! :)

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 2]
```diff
@@ -75,10 +77,14 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
...
-      })
-      router.push({ pathname: "/settings/my-collection" })
+      if (isEditing) {
+        router.push({ pathname: `/my-collection/artwork/${artworkId}` })
+      } else {
+        router.replace({
+          pathname: `/my-collection/artworks/${artworkId}/edit`,
+        })
+        router.push({ pathname: "/settings/my-collection" })
+      }
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:87]: Adjusting the redirect here.

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 3]
```diff
@@ -138,7 +144,11 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
                               py={2}
                             >
                               <BackLink
-                                to="/settings/my-collection"
+                                to={
+                                  isEditing
+                                    ? `/my-collection/artwork/${artwork.internalID}`
+                                    : "/settings/my-collection"
+                                }
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:151]: Adjusting the behaviour of the back button.

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:57]: +100

**Review by MounirDhahri — APPROVED** [review]

**Review by dariakoko — COMMENTED** [review]
👍

**Pull request merged**